packages/babel-parser/src/parser/expression.ts

```diff
@@ -3236,18 +3236,24 @@ export default abstract class ExpressionParser extends LValParser {
     this.expectPlugin("moduleBlocks");
     const node = this.startNode<N.ModuleExpression>();
     this.next(); // eat "module"
-    this.expect(tt.braceL);
+    if (!this.match(tt.braceL)) {
+      this.unexpected(null, tt.braceL);
+    }
+    // start program node immediately after `{`
+    const program = this.startNodeAt<N.Program>(
+      this.state.end,
+      this.state.endLoc,
+    );
+    this.next(); // eat `{`
```

Comment:
Nit, but we could use `this.state.lastTokEnd`/`this.state.lastTokEnd.index` to keep using expect instead of this math-createNode-next dance.

Reply:
They are not equivalent. For example
```js
module { /* 1 */ }
```
When we see a left brace `{`, `this.state.endLoc` points to the whitespace after `{`, while `this.state.lastTokEnd` after `{` is eaten will point to the whitespace immediately after the comment `/* 1 */`.

Reply:
👍